### PR TITLE
Fix examples of the scale_exponential expression function

### PR DESCRIPTION
**Pull request by DelazJ**
## Description

[Replace this with some text explaining the rationale and details about this pull request]

### CODE REVIEW

**Comment by agiudiceandrea** [comment]
I think it would be also useful to add such examples (and possibly also the corresponding ones for scale_polynomial) as tests in https://github.com/qgis/QGIS/blob/master/tests/src/core/testqgsexpression.cpp

**Comment by DelazJ** [comment]
> I think it would be also useful to add such examples (and possibly also the corresponding ones for scale_polynomial) as tests

Tests are already there: https://github.com/qgis/QGIS/commit/d0c7c6e9302e235240294598132d6efcbaf5aaeb#diff-78762f6fed0edd0a03a082ac901d682e7812bf3841a50ca205481f096d0f619f. Or do you mean these examples in particular?
I thought the function was younger than 3.34 but no, you are right, 3.34 shows the same unrelated examples.

**Comment by agiudiceandrea** [comment]
> do you mean these examples in particular?

Yes, I mean I would add:

```c++
      QTest::newRow( "scale_exponential(5,0,10,0,100,2)" ) << "scale_exponential(5,0,10,0,100,2)" << false << QVariant( 3.0303030303030303 );
      QTest::newRow( "scale_exponential(3,0,10,0,100,0.5)" ) << "scale_exponential(3,0,10,0,100,0.5)" << false << QVariant( 87.58553274682306 );
```
and the corresponding ones for scale_polynomial.

I think it would be a good practice to have the functions' examples also in the tests whenever it's possible.

File: tests/src/core/testqgsexpression.cpp [hunk 1]
```diff
@@ -1000,6 +1000,8 @@ class TestQgsExpression : public QObject
       QTest::newRow( "scale_linear(10,0,10,100,200)" ) << "scale_linear(10,0,10,100,200)" << false << QVariant( 200. );
       QTest::newRow( "scale_linear(-1,0,10,100,200)" ) << "scale_linear(-1,0,10,100,200)" << false << QVariant( 100. );
       QTest::newRow( "scale_linear(11,0,10,100,200)" ) << "scale_linear(11,0,10,100,200)" << false << QVariant( 200. );
+      QTest::newRow( "scale_linear(5,0,10,0,100,2)" ) << "scale_linear(5,0,10,0,100,2)" << false << QVariant( 50. );
+      QTest::newRow( "scale_linear(1500,1000,10000,9,20)" ) << "scale_linear(1500,1000,10000,9,20)" << false << QVariant( 9.61111111111111 );
```
Comment (agiudiceandrea) [inline tests/src/core/testqgsexpression.cpp:1004]:
```suggestion
      QTest::newRow( "scale_linear(5,0,10,0,100)" ) << "scale_linear(5,0,10,0,100)" << false << QVariant( 50. );
      QTest::newRow( "scale_linear(0.2,0,1,0,360)" ) << "scale_linear(0.2,0,1,0,360)" << false << QVariant( 72. );
      QTest::newRow( "scale_linear(1500,1000,10000,9,20)" ) << "scale_linear(1500,1000,10000,9,20)" << false << QVariant( 9.61111111111111 );
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**